ui/plugins/controller.js

```diff
@@ -373,71 +373,90 @@ treeherder.controller('PluginCtrl', [
-                                            $scope.$apply(thNotify.send("Request sent to backfill jobs", 'success'));
-                                        }, function (e) {
-                                            // The full message is too large to fit in a Treeherder
-                                            // notification box.
-                                            $scope.$apply(thNotify.send(ThTaskclusterErrors.format(e), 'danger', true));
-                                        });
+            if (!$scope.canBackfill()) {
+                return;
+            }
+            if (!$scope.user.loggedin) {
+                thNotify.send("Must be logged in to backfill a job", 'danger');
+                return;
+            }
+            if (!$scope.job.id) {
+                thNotify.send("Job not yet loaded for backfill", 'warning');
+                return;
+            }
+
+            if ($scope.job.build_system_type === 'taskcluster' || $scope.job.reason.startsWith('Created by BBB for task')) {
+                return ThResultSetStore.getGeckoDecisionTaskId(
+                    $scope.repoName,
+                    $scope.resultsetId).then(function (decisionTaskId) {
+                        return tcactions.load(decisionTaskId).then((results) => {
+                            const tc = thTaskcluster.client();
+                            const queue = new tc.Queue();
+                            const actionTaskId = tc.slugid();
+                            if (results) {
+                                const backfilltask = _.find(results.actions, {name: 'backfill'});
+                                // We'll fall back to actions.yaml if this isn't true
+                                if (backfilltask) {
+                                    const actionTask = tcactions.render(backfilltask.task, _.defaults({}, {
+                                        taskGroupId: decisionTaskId,
+                                        taskId: null,
+                                        task: null,
+                                        input: {},
+                                    }, results.staticActionVariables));
+
+                                    queue.createTask(actionTaskId, actionTask).then(function () {
+                                        $scope.$apply(thNotify.send("Request sent to backfill jobs", 'success'));
+                                    }, function (e) {
+                                        // The full message is too large to fit in a Treeherder
+                                        // notification box.
+                                        $scope.$apply(thNotify.send(ThTaskclusterErrors.format(e), 'danger', true));
                                     });
+                                }
+                            }
+
+                            // Otherwise we'll figure things out with actions.yml
```

Comment:
Is it worth having some kind of warning, maybe just console.log, so we can debug when this is occurring?